index.js

```diff
@@ -424,6 +424,39 @@
     };
   }
 
+  //  curry2 :: ((a, b) -> c) -> a -> b -> c
+  function curry2(f) {
+    return function(x) {
+      return function(y) {
+        return f (x, y);
+      };
+    };
+  }
+
+  //  curry3 :: ((a, b, c) -> d) -> a -> b -> c -> d
+  function curry3(f) {
+    return function(x) {
+      return function(y) {
+        return function(z) {
+          return f (x, y, z);
+        };
+      };
+    };
+  }
+
+  //  curry4 :: ((a, b, c, d) -> e) -> a -> b -> c -> d -> e
+  function curry4(f) {
+    return function(w) {
+      return function(x) {
+        return function(y) {
+          return function(z) {
+            return f (w, x, y, z);
+          };
+        };
+      };
+    };
+  }
```

Comment:
Are these functions used within the codebase? Why are we keeping them around?

Reply:
These functions are used to curry `Z` functions. `curry4` is applied exactly once: `curry4 (Z.lift3)`.